pkg/tsdb/prometheus/promclient/cache.go

```diff
@@ -48,8 +44,9 @@ func (c *ProviderCache) GetClient(headers map[string]string) (apiv1.API, error)
 }
 
 func (c *ProviderCache) key(headers map[string]string) string {
-	if c.jsonData.OauthPassThru {
-		return headers[authHeader] + headers[idTokenHeader]
+	var vals []string
+	for _, v := range headers {
+		vals = append(vals, v)
 	}
-	return noPassThrough
+	return strings.Join(vals, "")
```

Comment:
would the sort order of the map cause extra cache misses here?

> When iterating over a map with a range loop, the iteration order is not specified and is not guaranteed to be the same from one iteration to the next. If you require a stable iteration order you must maintain a separate data structure that specifies that order.

Reply:
Good catch. Just fixed this because it caused the test to fail